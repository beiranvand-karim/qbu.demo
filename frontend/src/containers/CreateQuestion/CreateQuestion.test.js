import {shallow} from 'enzyme'
import React from "react"
import {CreateQuestion} from "./"
import renderer from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom"

describe('<CreateQuestion/>', () => {

   const createQuestionErrorMock = jest.fn();
   const createQuestionSuccessMock = jest.fn();
   const createQuestionBeginMock = jest.fn();

   it('should render without crashing', () => {
      shallow(<CreateQuestion/>)
   });

   it('should create a question', () =>{
      window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve) => {
               resolve({
                  status: 200,
                  json: () => {
                     return new Promise((resolve) => {
                        resolve({
                           title: 'this is title',
                           text: 'this is text',
                           prize: 25
                        })
                     })
                  }
               })
            })
         }
      );

      const wrapper = shallow(<CreateQuestion
         createQuestionError={createQuestionErrorMock}
         createQuestionSuccess={createQuestionSuccessMock}
         createQuestionBegin={createQuestionBeginMock}/>);

      wrapper.find('Form').simulate('submit', {preventDefault() {}});

      expect(createQuestionBeginMock.mock.calls.length).toEqual(1);
      setImmediate(()=>{
         expect(createQuestionSuccessMock.mock.calls.length).toEqual(1);
      })

   });

   it('should fail to create a question', () => {

      window.fetch = jest.fn().mockImplementation(() => {
            return new Promise((reject) => {
               reject({
                  error: 'create question failed.'
               })
            })
         }
      );

      const wrapper = shallow(<CreateQuestion
         createQuestionError={createQuestionErrorMock}
         createQuestionSuccess={createQuestionSuccessMock}
         createQuestionBegin={createQuestionBeginMock}/>);

      wrapper.find('Form').simulate('submit', {preventDefault() {}});

      expect(createQuestionBeginMock.mock.calls.length).toEqual(2);
      setImmediate(()=>{
         expect(createQuestionErrorMock.mock.calls.length).toEqual(1);
      })

   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><CreateQuestion /></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});
