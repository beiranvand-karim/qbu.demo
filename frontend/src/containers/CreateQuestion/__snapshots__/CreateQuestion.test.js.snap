// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateQuestion/> should matches the snapshot 1`] = `
<div
  className="createQuestionFormContainer"
>
  <form
    className="createQuestionForm"
    onSubmit={[Function]}
  >
    <div
      className="formGroup position-relative form-group"
    >
      <label
        className=""
        htmlFor="title"
      >
        title
      </label>
      <input
        className="formGroup formGroup_input form-control"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="write a question title"
        type="text"
        value=""
      />
    </div>
    <div
      className="formGroup position-relative form-group"
    >
      <label
        className=""
        htmlFor="text"
      >
        text
      </label>
      <textarea
        className="formGroup formGroup_input form-control"
        id="text"
        name="text"
        onChange={[Function]}
        placeholder="write a description"
        rows={5}
        value=""
      />
    </div>
    <div
      className="formGroup position-relative form-group"
    >
      <label
        className=""
        htmlFor="prize"
      >
        prize
      </label>
      <input
        className="formGroup formGroup_input form-control"
        id="prize"
        name="prize"
        onChange={[Function]}
        placeholder="write a question title"
        type="text"
        value={0}
      />
    </div>
    <div
      className="formGroup position-relative form-group"
    >
      <label
        className=""
        htmlFor="options"
      >
        options
      </label>
      <input
        className="formGroup formGroup_input form-control"
        id="options"
        name="options"
        onChange={[Function]}
        placeholder="how many options?"
        type="text"
        value={4}
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
    <div>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</div>
`;
